allennlp/predictors/coref.py

```diff
@@ -53,6 +55,38 @@ def predict(self, document: str) -> JsonDict:
         """
         return self.predict_json({"document" : document})
 
+    def predict_from_list(self, tokenized_document: List[str]) -> JsonDict:
+        """
+        Predict the coreference clusters in the given document.
+
+        Parameters
+        ----------
+        tokenized_document : ``List[str``
+            A list of words representation of a tokenized document.
+
+        Returns
+        -------
+        A dictionary representation of the predicted coreference clusters.
+        """
+        return self.predict_words_list(tokenized_document)
+
+    def predict_words_list(self, words_list: List[str]) -> JsonDict:
+        instance = self._words_list_to_instance(words_list)
+        return self.predict_instance(instance)
+
+    def _words_list_to_instance(self, document_list: List[str]) -> Instance:
+        """
+        Create an instance from words list represent an already tokenized document,
+        for skipping tokenization when that information already exist for the user
+        """
+        spacy_document = self._spacy.tokenizer.tokens_from_list(document_list)
```

Comment:
It looks like you're calling the same spacy pipeline on the document twice here; once inside `tokenizer.tokens_from_list`, and then once again here.  I think all you need to do is call `spacy_document = self._spacy.tokens_from_list(words)` here (without adding the method onto the `SpacyWordSplitter`), then continue with this logic, and it should just work.  We don't need the extra method on `SpacyWordSplitter` at all.

Reply:
Here im using the actual Spacy tokenizer and not the WordSplitter one (which I use in srl), it will return a spacy.Doc object, I then run the pipeline on the doc object only once.

Reply:
Oh, I see - so yeah, you really don't need that extra method at all.